### PR TITLE
[vis tools] source selector

### PR DESCRIPTION
**Pull request by chejennifer**
- add source selector for vis tools
    - note: facet id is not remembered in the url hash because facet Ids can become stale

[screen-recording - 2023-09-11T173921.005.webm](https://github.com/datacommonsorg/website/assets/69875368/b43ca6ef-cd46-44e0-b8c0-d8f37b7e4fe6)
[screen-recording - 2023-09-11T174046.492.webm](https://github.com/datacommonsorg/website/assets/69875368/ade0a4d5-c045-4710-82f9-7783ed572fc4)
[screen-recording - 2023-09-11T174208.924.webm](https://github.com/datacommonsorg/website/assets/69875368/db4904cd-923d-4656-b365-fb866f89c2ae)

### CODE REVIEW

File: static/js/utils/data_fetch_utils.ts [hunk 1]
```diff
@@ -196,12 +209,101 @@ export function getSeriesWithin(
...
+            result[svDcid][obs.facet] = respData.facets[obs.facet];
+          }
+        });
+      }
+      return result;
+    });
+}
+
+/**
+ * Gets the available favets for each variable given a list of entities.
```
Comment (shifucun) [inline static/js/utils/data_fetch_utils.ts:273]: s/favets/facets

**Review by shifucun — COMMENTED** [review]
This is very cool! I would suggest adding one webdriver test for each tool, and make sure facets can be selected and charts can show up!

File: static/js/utils/data_fetch_utils.ts (hunk 1 above)
Comment (chejennifer) [reply static/js/utils/data_fetch_utils.ts]: done

**Comment by chejennifer** [comment]
> This is very cool! I would suggest adding one webdriver test for each tool, and make sure facets can be selected and charts can show up!

thanks for the review! I agree with adding webdriver tests, but maybe in a separate PR?

**Review by shifucun — APPROVED** [review]

**Pull request merged**